### PR TITLE
add extension function and extension property for getting logger

### PR DESCRIPTION
**Pull request by LiYing2010**
this PR fix issue #77

change list:
add extension function and extension property for getting logger, to simplify the way how to define logger

with the help of this new extension property `T.log`, we can use `log` in any class, without calling the logger initialization code.

Before:
we need this logger initialization code:
`private val logger = KotlinLogging.logger {}`
before we can output log message

After this change, we can call
```
log.debug { "some message" }
```

without the logger initialization code.

for a large project, there could be many many classes, so this new extension property can save much of the duplicated code

### CODE REVIEW

**Comment by sghpjuikit** [comment]
Hello, I'm passing by and see the PR and wanted to chip in my opinion. While I see the good intention of the author of this PR, I am very against merging this into the library.

- This makes the library inconsistent in how it wants users to use logger by providing two separate mechanisms.
- This provides extension on Any, which pollutes API and should only be provided if merits vastly overshadow demerits. Not the case here, I'm afraid
- The design decision to go with extension (that is 'implicitly' invoked on `this`) is bad for the same reason scala implicit conversions are (not always) bad - everything is everything and nothing is clear. This basically implicitly (with just import) turns every type into some kind of Loggable. It is debatable whether this is clean or clean-er as other approaches, as with proper mixins (like in scala) the method has its elegance. However for logging this is, clear case of no. In fact it is very bad and prone to cause unexpected behavior. See below.
- The design decision to abandon companions for convenience is a trap. Logging is cross cutting concern and logger belongs to the class instance, not the object. Companion is the correct place to define logger in and provide it for instances. The provided extension is a sugar to call the 'companion' logger without the companion (presumably by tapping right into the global logger map) - like Java member method that secretly calls static method instead). Ignoring the discussion if this is ever desirable (convenience that abandons correctness, due to added cognitive load for the user), this is inherently dangerous:

1 The receiver the log/logger() is called on is often not clear and sometimes will result in wrong one. Consider using it in nested typesafe builder and other constructs that change semantics of this within scope. Which this do you want to be used to log? Expecting developer to pay attention to these cases in irresponsible and opens then up to nasty problems that could be avoided easily. How? By specifying the exact class the used logger is bound to by binding it to the companion object!

2 How do you expect this to work in subclasses where user would declare another logger for companion. In this case your approach using `this` instead of `this.super` or something like that would result in wrong logger instance to be used again. 

TLDR: the logger should be context bound, always, at creation time, explicitly. It gives clear semantics and expected behaviot with 0 pitfalls.

How do I know all this from top of my head? I myself used this very approach in the past, having a static method (Java) I could call like this: `log(this).error(...)`.

**Comment by oshai** [comment]
Will not merge, see issue #77
